server/embed/etcd.go

```diff
@@ -774,3 +804,49 @@ func parseCompactionRetention(mode, retention string) (ret time.Duration, err er
 	}
 	return ret, nil
 }
+
+func (e *Etcd) setupTracing(ctx context.Context) (tracesdk.SpanExporter, *tracesdk.TracerProvider, error) {
+	exporter, err := otlp.NewExporter(ctx,
+		otlpgrpc.NewDriver(
+			otlpgrpc.WithEndpoint(e.cfg.ExperimentalDistributedTracingAddress),
+			otlpgrpc.WithInsecure(),
+		))
+	if err != nil {
+		return nil, nil, err
+	}
+	res := resource.NewWithAttributes(
+		semconv.ServiceNameKey.String(e.cfg.ExperimentalDistributedTracingServiceName),
+	)
+	// As Tracing service Instance ID must be unique, it should
+	// never use the empty default string value, so we only set it
+	// if it's a non empty string.
+	if e.cfg.ExperimentalDistributedTracingServiceInstanceID != "" {
+		resWithIDKey := resource.NewWithAttributes(
+			(semconv.ServiceInstanceIDKey.String(e.cfg.ExperimentalDistributedTracingServiceInstanceID)),
+		)
+		// Merge resources to combine into a new
+		// resource in case of duplicates.
+		res = resource.Merge(res, resWithIDKey)
+	}
+
+	tp := tracesdk.NewTracerProvider(
+		tracesdk.WithBatcher(exporter),
+		tracesdk.WithResource(res),
+	)
+
+	otel.SetTextMapPropagator(
```

Comment:
I am guessing we need this for apiserver (or other components that would also add this kind of tracing) as you mentioned above, or is there a way to avoid that?

Reply:
You can plumb it through to the interceptors, and pass it as an argument: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/13d72c9c8dda39280ead2e291cbed8f51a6f65f9/instrumentation/google.golang.org/grpc/otelgrpc/grpctrace.go#L67

Reply:
Basically do the same thing you did with the TracerProvider